### DISCUSSION TITLE
Allow ```var``` in method overrides

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7731] [General]
Hi C# team:
When working with Overrides, it would be great if I could use ```var```.  This would save me a lot of manual refactoring work which changing types.  ```var``` should work whenever it can be absolutely resolved with no ambiguity.  If var would introduce an ambiguity in an override, then it should not be allowed in that situation.

Here is an example:

```csharp
//Base class must have explicit types
public abstract class ExampleBase<TInput, TKey, TValue> {

  public abstract KeyValuePair<TKey, TValue> GetValues(TInput Input);

}

//Derived class can then use var
public class Example : ExampleBase<Person, string, string> {

  public override var GetValues(var Input) {
    var ret = KeyValuePair.Create("Name", Input.Name);
    return ret;
  }

}

```

**Comment by HaloFour** [comment]
The IDE already generates the signature for you so I don't see much point in avoiding having the explicit types in the signatures.  I'd rather the clarity than having to navigate back to the base type to understand what it does.  Not to mention, that wouldn't work with overloads.

**Reply by yaakov-h** [reply]
Why wouldn’t it work with overloads? You can’t overload methods on return type only.

**Reply by HaloFour** [reply]
This proposal is also using `var` for the parameters.

**Reply by yaakov-h** [reply]
oh I missed that detail, thanks

**Comment by 3fbaea00** [comment]
I'm against it. 
But about your idea, I think override don't need even var, it anyway don't give any information about return type.
With the same logic, you can remove the modifier, leaving only keyword `override`:
```
override GetValues(NotVarTypeValue Input) {
    var ret = KeyValuePair.Create("Name", Input.Name);
    return ret;
  }
```
Probably devs want to leave information about method, so they don’t take this approach

**Comment by huoyaoyuan** [comment]
With covariant return, the return type of overridden method isn't necessarily the same of base method.
`var` also introduces overload resolution hell if two virtual methods are having the same name.
